src/integrations/prefect-docker/tests/deployments/test_steps.py

```diff
@@ -501,3 +501,22 @@ def test_uncached_push_docker_image(self, mock_docker_client):
         assert mock_docker_client.login.call_count == 3
         expected_push_calls = 1 + len(additional_tags)
         assert mock_docker_client.api.push.call_count == expected_push_calls * 3
+
+    def test_avoids_aggressive_caching(self, mock_docker_client):
+        image_name = "registry/repo"
+        tag = "latest"
+        credentials = {"username": "user", "password": "pass"}
+
+        build_docker_image(
+            image_name=image_name,
+            tag=tag,
+        )
+
+        # Push the image (this should not hit the cache)
+        push_docker_image(
```

Comment:
wouldn't the first call of this _always_ avoid the cache?

Reply:
ah it would because of the inclusion of `credentials`

but otherwise no, and this is the bug I was trying to address. it was only based on the parameters and not the function name

will update

Reply:
sorry I think I'm just misunderstanding but I still don't get it; calling a function for the first time should never use a cache, so I don't understand how this test tests anything?

Reply:
Caught up - the problem was that these functions were sharing a cache which is clearly problematic! 👍